src/Symfony/Component/Uid/ArgumentResolver/UidValueResolver.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Uid\ArgumentResolver;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Uid\AbstractUid;
+use Symfony\Component\Uid\Attribute\Uid;
+
+final class UidValueResolver implements ArgumentValueResolverInterface
+{
+    private string $defaultFormat;
+
+    public function __construct(string $defaultFormat = Uid::FORMAT_CANONICAL)
+    {
+        $this->defaultFormat = $defaultFormat;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(Request $request, ArgumentMetadata $argument): bool
+    {
+        if ($argument->isVariadic() || !\is_string($request->attributes->get($argument->getName())) || null === $argument->getType()) {
+            return false;
+        }
+
+        try {
+            $refl = new \ReflectionClass($argument->getType());
+        } catch (\ReflectionException) {
+            return false;
+        }
+
+        return !$refl->isAbstract() && $refl->isSubclassOf(AbstractUid::class);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function resolve(Request $request, ArgumentMetadata $argument): iterable
+    {
+        $format = $this->defaultFormat;
+        if (null !== $configuration = method_exists($argument, 'getAttributes') ? $argument->getAttributes(Uid::class, ArgumentMetadata::IS_INSTANCEOF)[0] ?? null : null) {
```

Comment:
why the method_exists check? Can't we remove it?

Reply:
I borrowed that part from https://github.com/symfony/symfony/pull/43854. From what I understood, it's to be compatible with symfony/http-kernel 4.4 but I guess we don't really care?

Reply:
no need for that yes, maybe we can clean the call from other places if that's legacy